### PR TITLE
First Draft of Kwil Concepts

### PR DESCRIPTION
**Pull request by KwilLuke**
A first draft of Kwil Concepts section.

The goal of Kwil Concepts is to provide a single entry point that someone who is new to Kwil can start from and obtain the base knowledge to be effective throughout our system (building applications, deploying a database, starting a network, etc).

I created this simple diagram to try to explain where each concept is relevant, if possible:
Kwil Concepts

If you have feedback on the diagram, feel free to comment on the [LucidChart](https://lucid.app/lucidchart/ef3367eb-427d-44ca-800e-9b8523644286/edit?viewport_loc=1661%2C-120%2C1999%2C1479%2Cm-5o7ONTd-nK&invitationId=inv_4f2e8cc8-0547-4362-8a3b-a5165496e2bb) or leave comments directly in this pull request.

### CODE REVIEW

File: docs/concepts.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+In Kwil, databases refer to SQL deployments on a Kwil Network. The Kwil [SQL engine](/docs/category/sql-as-understood-by-kwil) is based on SQLite with modifications to support deterministic execution and consensus.
+
+### Kuneiform
+
+[Kuneiform](/docs/category/kuneiform) is the syntax language for defining a database on Kwil. You can think of Kuneiform as a smart contract language (e.g. [Solidity](https://soliditylang.org/)), but for relational databases.
+
+In a Kuneiform file, you can define:
+- Relational Tables
+- Actions - CRUD Operations that can be executed on the database.
+- Extensions - Custom logic that can be executed on the database.
```
Comment (Yaiba) [inline docs/concepts.mdx:37]: This is not accurate, developer can only import/config an existing Extension, and use extension in Action, in Kuneiform.    
The definition of the Extension right now is in separate docker

File: docs/concepts.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+### Database Identifiers
+
+Database identifiers (DBIDs) are unique identifiers for a database on a [Kwil Network](#kwil-network). DBIDs are generated from the public key of the account that deployed the database and the database name. A DBID is a SHA224 hash of the database owner's account public key and database name.
```
Comment (Yaiba) [inline docs/concepts.mdx:41]: I guess this is nbd, the hash is `hash(name+public_key)`, so maybe change to `hash of database name and ...key`

File: docs/concepts.mdx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+### Transactions
+
+Transactions are database deployments, database drops, and state changing (Create, Update, Delete) actions that are executed on a database. Transactions are signed by an account and are propagated to the Kwil Network. Transactions are executed deterministically on each node in the network and are validated by the nodes. 
+
+### View Actions
+
+View actions are read-only actions that can be executed on a database. View actions are called from a single node and may or may not require signature. The advantage of view actions is that they can be executed without the need to wait for consensus.
```
Comment (Yaiba) [inline docs/concepts.mdx:49]: I think `view action` has no advantage imo, it's required feature for reading data(this is the only way)

File: docs/concepts.mdx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+## Kwil Node
+
+Kwil Nodes run the [Kwil Daemon](/docs/category/kwil-daemon) and are the infrastructure that power Kwil Networks. Kwil Nodes are responsible for:
+- Storing the state of the Kwil Network
+- Executing transactions
+- Validating transactions
+- Propagating transactions to other nodes
+- Maintaining consensus
+
+Each node has a public / private ED25519 keypair that is used to sign transactions and blocks. Note that the Node keypair is only for validating the blockchain, and is different from the [account](#accounts) keypair.
```
Comment (Yaiba) [inline docs/concepts.mdx:64]: Should we only mention secp156k1 key in Account section?

**Review by Yaiba — APPROVED** [review]

File: docs/concepts.mdx (hunk 4 above)
Comment (KwilLuke) [reply docs/concepts.mdx:64]: I think we still support ed25519 out of the box. @brennanjl can you confirm?

Comment (brennanjl) [reply docs/concepts.mdx:64]: yes we do

File: docs/concepts.mdx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+<div style={{ display: "flex", justifyContent: "center" }}>
+    <img src={require('../static/img/kwil-concepts.png').default} alt="Concepts Overview" />
+</div>
+
+## Accounts
+
+Accounts are public / private keypairs that represent a single user or entity on a Kwil Network. Accounts are used to deploy databases and execute CRUD operations.
+
+Each account has a public_key, nonce, and balance.
```
Comment (brennanjl) [inline docs/concepts.mdx:22]: public_key should be public key

File: docs/concepts.mdx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+Accounts are identified by public keys. Natively, Kwil supports Secp256k1 keys (used in EVM) and ED25519 keys (used in Solana, NEAR, etc).
+
+## Databases 
+
+In Kwil, databases refer to SQL deployments on a Kwil Network. The Kwil [SQL engine](/docs/category/sql-as-understood-by-kwil) is based on SQLite with modifications to support deterministic execution and consensus.
```
Comment (brennanjl) [inline docs/concepts.mdx:28]: rreplace "SQL deployments" with "relational database schemas"

File: docs/concepts.mdx (hunk 3 above)
Comment (brennanjl) [reply docs/concepts.mdx:49]: I think its good.  However, I would add "View actions are the primary way of reading data from a database"

Comment (KwilLuke) [reply docs/concepts.mdx:49]: Maybe the only way? Assuming we are getting rid of the query endpoint eventually.

**Pull request merged**